tests/phpunit/tests/query/cacheResults.php

```diff
@@ -171,6 +171,58 @@ public function test_generate_cache_key_placeholder() {
 		$this->assertSame( $cache_key_1, $cache_key_2, 'Cache key differs when using wpdb placeholder.' );
 	}
 
+	/**
+	 * @covers WP_Query::generate_cache_key
+	 * @ticket 59442
+	 */
+	public function test_generate_cache_key_unregister_post_type() {
+		global $wpdb;
+		register_post_type(
+			'wptests_pt',
+			array(
+				'exclude_from_search' => false,
+			)
+		);
+		$query_vars = array(
+			'post_type' => 'any',
+		);
+		$fields     = "{$wpdb->posts}.ID";
+		$query1     = new WP_Query( $query_vars );
+		$request1   = str_replace( $fields, "{$wpdb->posts}.*", $query1->request );
+
+		$reflection = new ReflectionMethod( $query1, 'generate_cache_key' );
+		$reflection->setAccessible( true );
+
+		$cache_key_1 = $reflection->invoke( $query1, $query_vars, $request1 );
+		unregister_post_type( 'wptests_pt' );
+		$cache_key_2 = $reflection->invoke( $query1, $query_vars, $request1 );
+
+		$this->assertNotSame( $cache_key_1, $cache_key_2, 'Cache key should differs when after unregister post type.' );
+	}
+
+	/**
+	 * @ticket 59442
+	 *
+	 * @covers WP_Query::generate_cache_key
+	 *
+	 * @dataProvider data_query_cache_duplicate
+	 */
+	public function test_generate_cache_key_normalize( $query_vars1, $query_vars2 ) {
+		global $wpdb;
+
+		$fields   = "{$wpdb->posts}.ID";
+		$query1   = new WP_Query( $query_vars1 );
+		$request1 = str_replace( $fields, "{$wpdb->posts}.*", $query1->request );
+
+		$reflection = new ReflectionMethod( $query1, 'generate_cache_key' );
+		$reflection->setAccessible( true );
+
+		$cache_key_1 = $reflection->invoke( $query1, $query_vars1, $request1 );
+		$cache_key_2 = $reflection->invoke( $query1, $query_vars2, $request1 );
```

Comment:
It's not yet defined but shouldn't this be using the request value from the second query.

Reply:
I tried to do this in a early versions of the patch. It just made the tests harder to follow. I think this is fine. We can improve the tests later.